src/main/java/MongoDBDAO.java

```diff
@@ -1,139 +1,140 @@
-import com.mongodb.ConnectionString;
-import com.mongodb.MongoException;
+import com.mongodb.*;
 import com.mongodb.client.*;
 import org.bson.Document;
 import org.bson.types.ObjectId;
-
-import java.text.DateFormatSymbols;
+import java.rmi.ServerException;
 import java.util.Calendar;
 import java.util.Date;
-import java.util.InputMismatchException;
-
 import static com.mongodb.client.model.Filters.*;
 import static com.mongodb.client.model.Projections.*;
 
 public class MongoDBDAO implements WeatherstationDAO {
 
+
+
     @Override
-    public String InsertToDatabase(String jsonMessage) {
-        MongoCollection<Document> collection = MongoConnection.getInstance("AgricircleDB").getDatabase().getCollection("Weatherstation1");
+    public String setWeatherstationData(String jsonMessage, String weatherstationName) throws ServerException {
+        //For Production:
+        //MongoCollection<Document> collection = MongoConnection.getInstance("AgricircleDB").getDatabase().getCollection(weatherstationName);
 
+        //For Development:
+        MongoCollection<Document> collection = MongoConnection.getInstance("AgricircleDB").getDatabase().getCollection("Weatherstation1");
         try {
             collection.insertOne(Document.parse(jsonMessage));
             return "Database updated";
-        } catch (MongoException mwe) {
-            //  Block of code to handle error
-            return mwe.getMessage();
-            //throw new DBException("Soe")
+        } catch (MongoWriteException | MongoWriteConcernException e) {
+            throw new ServerException(e.getMessage());
```

Comment:
Subclass your own exception - Don't use rmi.ServerException

Reply:
I'm not sure what you mean by 'Subclass your own exception'
Could you elaborate?